Index: ejbca_ce_6_3_1_1/src/java/org/ejbca/util/keystore/PasswordCallBackHandler.java
===================================================================
--- ejbca_ce_6_3_1_1.orig/src/java/org/ejbca/util/keystore/PasswordCallBackHandler.java
+++ ejbca_ce_6_3_1_1/src/java/org/ejbca/util/keystore/PasswordCallBackHandler.java
@@ -13,6 +13,7 @@
 package org.ejbca.util.keystore;
 
 import java.io.IOException;
+import java.util.Scanner;
 
 import javax.security.auth.callback.Callback;
 import javax.security.auth.callback.CallbackHandler;
@@ -37,7 +38,12 @@ public class PasswordCallBackHandler imp
 				PasswordCallback pc = (PasswordCallback)callbacks[i];
 				System.err.print(pc.getPrompt());
 				System.err.flush();
-				pc.setPassword(System.console().readPassword());
+				if (System.console() != null){
+				    pc.setPassword(System.console().readPassword());
+                                } else {
+                                   final Scanner sc = new Scanner (System.in);
+                                   pc.setPassword(sc.next().toCharArray());
+                                }
 
 			} else {
 				throw new UnsupportedCallbackException
